lib/std/private/ntpath.nim

```diff
@@ -0,0 +1,64 @@
+# This module contains derivative work of Python's `ntpath.py` module. This is
+# attributed to the PSF-2.0 license with the given notice:
+#
+# Copyright (c) 2001-2022 Python Software Foundation; All Rights Reserved
```

Comment:
This makes me nervous... :-)

The PSF does not hold the rights on this Nim code that was only inspired by some Python code. As far as I can tell.

Reply:
After reading this: https://softwareengineering.stackexchange.com/a/234829, I decided to add this copyright notice. The `splitdrive` function in Python is PSF-2.0-licensed as far as I understood the Python license. PSF-2.0 is a BSD-style license.

Also, `lib/std/private/win_setenv.nim` in Nim has a copyright notice to Facebook at the top. I don't have an interest in licensing, so I would like to be wrong. Should I just remove the copyright stuff in ntpath.nim and tntpath.nim?

Reply:
While do I think that some form of attribution should be given, I don't feel that code would fall under the PSF's copyright. This is mainly because the `splitDrive` algorithm is fairly "obvious" - there's only so many ways one could implement such logic. Additionally, I've seen this kind of thing before, where an algorithms in one project contains attribution to the same algorithm in another project; sometimes the two projects are written in the same language, sometimes they are written in another language.

Of course, if the PSF comes along and says, "that code is a derivative work, and should be under the PSF license", then we can discuss that with them. Until then, a simple "inspired by <url to file>" is good enough.

Reply:
Thanks for the explanation. I dropped the attribution to Python.